core/intr_exc.c

```diff
@@ -218,20 +218,16 @@ void hax_inject_exception(struct vcpu_t *vcpu, uint8_t vector, uint32_t error_co
         if (error_code != NO_ERROR_CODE) {
             intr_info |= 1 << 11;
             if (vector == VECTOR_PF) {
-                vcpu->vmcs_pending_entry_error_code = 1;
-                vmx(vcpu, entry_exception_error_code) = error_code;
+                vmcs_write(vcpu, VMX_ENTRY_EXCEPTION_ERROR_CODE, error_code);
             } else {
                 vmwrite(vcpu, VMX_ENTRY_EXCEPTION_ERROR_CODE, error_code);
```

Comment:
Use `vmcs_write` instead?

Reply:
In that case, I should get rid of the `if (vector == VECTOR_PF)` altogether... I wonder why it was placed there to begin with.